Link store address to directions in maps app

wangannie opened the pull request:
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Tap the address line of a store to either:
  * Get directions in an external maps app
  * Copy the address to clipboard

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links

### Online sources
[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
n/a
### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
n/a
## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
* Make sure to set your simulator location to be near the DC area (otherwise it might open similarly-named stores...not sure how to fix this)
![image](https://user-images.githubusercontent.com/21160510/77599501-19930e00-6ec2-11ea-8578-bc37f2c16f47.png)

* Try opening directions for a bunch of different stores—it searches for the nearest store by address closet to the store's longitude, there may be some exceptions where it doesn't find the right store. I ended up abandoning longitude/latitude bc it was really jank for Android, so I went with a query string that may be hit or miss.
## Next steps
[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

* Look into [Action Sheets](https://github.com/expo/react-native-action-sheet) instead of Alerts 
## Tests Performed, Edge Cases
[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

* Tried to simulate cross-platform compatibility by setting it to Google maps for Android 
### Screenshots
https://www.loom.com/share/836be07d58664c77a39bdd62dfdc05de
![image](https://user-images.githubusercontent.com/21160510/77599408-c9b44700-6ec1-11ea-9d03-cdfba3159f92.png)
![image](https://user-images.githubusercontent.com/21160510/77599413-cf119180-6ec1-11ea-9c0f-6285f19c0b3c.png)


[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

tommypoa commented on components/store/StoreCard.js:
```diff
@@ -51,14 +56,41 @@ function StoreCard({ store, callBack, seeProduct }) {
+              {
+                text: 'Get Directions',
+                onPress: () =>
+                  openMap({
```
I tested this on my phone (iOS) but because I had my default Maps app deleted, it prompted me to download the default app and is not able to have it open on other map apps (GoogleMaps, Waze etc.). Am not sure if this will be a big problem (I might be the rare few who deletes the default Map app) but just to note the edge case! 

As a reference, the `react-native-open-map` package (https://www.npmjs.com/package/react-native-open-map) might give some additional flexibility if we want to account for this case.

wangannie replied:
WOW thank you `react-native-map-link`(what that's based off of) ended up being exactly what I needed! Wish I had found this earlier.

wangannie commented:
Updates:
* Switched to a different package `react-native-map-link` for better cross-platform compatibility (and to accommodate those who uninstall their system default map apps).
![image](https://user-images.githubusercontent.com/21160510/77617454-778b1a00-6ef1-11ea-89cf-daed7a523d13.png)
* Moved copy to clipboard to work on a long press instead  of a click (didn't want too many nested system dialogs)
![image](https://user-images.githubusercontent.com/21160510/77617605-d51f6680-6ef1-11ea-9a1a-a72b3d630b67.png)
* Cleaned everything up into helper functions based on @tommypoa 's  suggestion

kennethlien commented on components/store/StoreCard.js:
```diff
@@ -19,8 +20,24 @@ import StoreProductButton from './StoreProductButton';
+
+  const writeAddressToClipboard = () => {
+    Clipboard.setString(address);
+    Alert.alert('Copied to Clipboard!', address);
```
Could this be a popup that just fades away? It seems less obtrusive.
